test/brotli.spec.ts

```diff
@@ -68,4 +68,151 @@ describe("Brotli-wasm", () => {
+    });
+
+    it("does not fail when streamingly compressing with an illegal quality value", () => {
+        const input = Buffer.from("Test input data");
+        const input1 = input.slice(0, input.length / 2);
+        const input2 = input.slice(input.length / 2);
+        const quality = 12;
+        const stream = new brotli.CompressStream(quality);
+        const output1 = stream.compress(input1, 100);
+        expect(stream.result()).to.equal(brotli.BrotliStreamResult.NeedsMoreInput);
+        const output2 = stream.compress(input2, 100);
+        expect(stream.result()).to.equal(brotli.BrotliStreamResult.NeedsMoreInput);
+        const output3 = stream.compress(undefined, 100);
+        expect(stream.result()).to.equal(brotli.BrotliStreamResult.ResultSuccess);
+        expect(Buffer.concat([output1, output2, output3]).toString('base64')).to.equal('Gw4A+KWpyubolCCjVAjmxJ4D');
+    });
+
+    it("cleanly fails when streamingly decompressing garbage", () => {
+        const input = Buffer.from("This is not brotli data, it's just a string");
+        const stream = new brotli.DecompressStream();
+        expect(() =>
+            stream.decompress(input, 100)
+        ).to.throw('Brotli streaming decompress failed');
+        expect(stream.result()).to.lt(0);
+    });
+
+    it("can compress & decompress back to the original result", () => {
+        const s = "Some thrilling text I urgently need to compress";
+        const encInput = Buffer.from(s);
+        const encInput1 = encInput.slice(0, encInput.length / 2);
+        const encInput2 = encInput.slice(encInput.length / 2);
+        const encStream = new brotli.CompressStream();
+        const encOutput1 = encStream.compress(encInput1, 100);
+        expect(encStream.result()).to.equal(brotli.BrotliStreamResult.NeedsMoreInput);
+        const encOutput2 = encStream.compress(encInput2, 100);
+        expect(encStream.result()).to.equal(brotli.BrotliStreamResult.NeedsMoreInput);
+        const encOutput3 = encStream.compress(undefined, 100);
+        expect(encStream.result()).to.equal(brotli.BrotliStreamResult.ResultSuccess);
+        const encOutput = Buffer.concat([encOutput1, encOutput2, encOutput3]);
+
+        const decInput1 = encOutput.slice(0, encOutput.length / 2);
+        const decInput2 = encOutput.slice(encOutput.length / 2);
+        const decStream = new brotli.DecompressStream();
+        const decOutput1 = decStream.decompress(decInput1, 100);
+        expect(decStream.result()).to.equal(brotli.BrotliStreamResult.NeedsMoreInput);
+        const decOutput2 = decStream.decompress(decInput2, 100);
+        expect(decStream.result()).to.equal(brotli.BrotliStreamResult.ResultSuccess);
+        const decOutput = Buffer.concat([decOutput1, decOutput2]);
+
+        expect(decOutput.toString('utf8')).to.equal(s);
+    });
+
+    it("can compress & decompress back to the original result with a different quality setting", () => {
```

Comment:
For both these two tests, can you put 'streaming' in the name somewhere? Currently they're the same copy pasted name from the existing non-streaming equivalent.